### PR TITLE
feature: Header styling

### LINKED ISSUE #187: app title bar / header
link to allencell.org, and help menu
"Load" button not required for March release?

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
This closes #187, copying over the Header implementation from colorizer.

*Estimated review size: small, 15-20 minutes (faster when comparing with copied code)*

## Changes:
- Copies the Header implementation from `nucmorph-colorizer`.
- Adds SVG logos with a white fill for the dark background.
- Adds default styling for "link"-type buttons, and adds two placeholder buttons, Load + Share.
  - Buttons have been disabled until they are implemented.

image

Hovering buttons:
image

### CODE REVIEW

File: src/website/assets/icons/dropdown-arrow.svg
Comment (ShrimpCryptid) [inline src/website/assets/icons/dropdown-arrow.svg:1]: Currently unused! I need to copy over the Help and AccessibleDropdown components still, but I'm saving it for a later PR so this one doesn't get too much larger.

File: src/website/components/Header.tsx
Comment (ShrimpCryptid) [inline src/website/components/Header.tsx:1]: Compare with https://github.com/allen-cell-animated/nucmorph-colorizer/blob/8caf8ec22456ee28b1ac97a2aad0393c975c28f3/src/components/Header.tsx if desired. Component should be nearly identical with a few minor changes.

List of changes:
- L9-13: Added a constant for the header height.
- Edited the size of the logo based on designs.
- Added styling rules for the app title.
- Edited styling variable names to match definitions in `StyleProvider`.

File: src/website/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      display: none;
+    }
+
+    & > div > svg:last-child {
+      display: block;
+      visibility: visible;
+    }
+  }
+`;
+
```
Comment (interim17) [inline src/website/components/Header.tsx:42]: There is a mid-width where the logos switch over and the load and share buttons seem to get mis-positioned, as the screen size drops lower they go back to where they are supposed to be.
Screenshot 2024-04-16 at 11 58 42 AM

**Review by interim17 — COMMENTED** [review]
Generally looking good, leaving a question about resize behavior

File: src/website/components/Header.tsx (hunk 1 above)
Comment (ShrimpCryptid) [reply src/website/components/Header.tsx:42]: Yup, I need to tweak the magic numbers a bit for this viewer :X 
I haven't added the help dropdown yet either so I'll fix it for now but add a TODO.

Comment (ShrimpCryptid) [reply src/website/components/Header.tsx:43]: Fixed!

**Review by interim17 — APPROVED** [review]

**Comment by meganrm** [comment]
@ShrimpCryptid  can you resolve the conflicts?

**Comment by ShrimpCryptid** [comment]
> @ShrimpCryptid can you resolve the conflicts?

Waiting on the base branch which is also under review... The changes are just file moves.

File: src/website/components/LandingPage/index.tsx [hunk 2]
```diff
@@ -291,13 +293,22 @@ export default function LandingPage(): ReactElement {
   }, []);
 
   return (
-    <div style={{ backgroundColor: "var(--color-landingpage-bg)", height: "100%" }}>
-      {/* <Header>
+    <div style={{ backgroundColor: "var(--color-landingpage-bg)", minHeight: "100%" }}>
```
[the remark this answers is not in the document]
Comment (ShrimpCryptid) [reply src/website/components/LandingPage/index.tsx:296]: No strong reasoning! I tend to do inline if it's a relatively simple style (1-2 style overrides), any more than that and it should probably have its own styled component.

[the remark this answers is not in the document]
Comment (ShrimpCryptid) [reply src/website/components/LandingPage/index.tsx:296]: It also depends on if the styling is reused, or if it's particularly important (which you could certainly make the argument for here)

**Review by meganrm — APPROVED** [review]

**Pull request merged**